src/controllers/__tests__/pagoPAController.test.ts

```diff
@@ -0,0 +1,240 @@
+};
+
+const proxyUserResponse: PagoPAUser = {
+  email: aSpidEmailAddress,
+  family_name: aValidFamilyname,
+  mobile_phone: aMobilePhone,
+  name: aValidName
+};
+
+const proxyExtendedUserResponse: ExtendedPagoPAUser = {
+  family_name: aValidFamilyname,
+  fiscal_code: aFiscalNumber,
+  mobile_phone: aMobilePhone,
+  name: aValidName,
+  notice_email: aCustomEmailAddress,
+  spid_email: aSpidEmailAddress
+};
+
+const anIsInboxEnabled = true as IsInboxEnabled;
+const anIsWebookEnabled = true as IsWebhookEnabled;
+const aPreferredLanguages: ReadonlyArray<PreferredLanguage> = [
+  PreferredLanguageEnum.it_IT
+];
+
+const userInitializedProfile: InitializedProfile = {
+  email: aCustomEmailAddress,
+  family_name: aValidFamilyname,
+  fiscal_code: aFiscalNumber,
+  has_profile: true,
+  is_email_enabled: true,
+  is_email_validated: true,
+  is_inbox_enabled: anIsInboxEnabled,
+  is_webhook_enabled: anIsWebookEnabled,
+  name: aValidName,
+  preferred_languages: aPreferredLanguages,
+  spid_mobile_phone: aMobilePhone,
+  version: 42
+};
+
+const badRequestErrorResponse = {
+  detail: expect.any(String),
+  status: 400,
+  title: expect.any(String),
+  type: undefined
+};
+
+const mockGetProfile = jest.fn();
+jest.mock("../../services/profileService", () => {
+  return {
+    default: jest.fn().mockImplementation(() => ({
+      getProfile: mockGetProfile
+    }))
+  };
+});
+
+describe("PagoPaController#getUser", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("calls the getUser on the PagoPaController with valid values", async () => {
+    const req = mockReq();
+
+    // tslint:disable-next-line: no-object-mutation
+    req.user = mockedUser;
+
+    const api = new ApiClientFactory("", "");
+    const profileService = new ProfileService(api);
+    const pagoPAController = new PagoPAController(profileService);
+
+    const response = await pagoPAController.getUser(req);
+    expect(response).toEqual({
+      apply: expect.any(Function),
+      kind: "IResponseSuccessJson",
+      value: proxyUserResponse
+    });
+  });
+
+  it("should return an error if spid_email is not available for the user", async () => {
+    const req = mockReq();
+    const res = mockRes();
+
+    const mockedUserWithoutSpidEmail = {
+      ...mockedUser,
+      spid_email: undefined
+    };
+    // tslint:disable-next-line: no-object-mutation
+    req.user = mockedUserWithoutSpidEmail;
+
+    const api = new ApiClientFactory("", "");
+    const profileService = new ProfileService(api);
+    const pagoPAController = new PagoPAController(profileService);
+
+    const response = await pagoPAController.getUser(req);
+    response.apply(res);
+    expect(res.json).toBeCalledWith(badRequestErrorResponse);
+  });
+});
+
+describe("PagoPaController#getExtendedUser", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("calls the getExtendedUser on the PagoPaController with valid values", async () => {
```

Comment:
since "calls the getExtendedUser on the PagoPaController" is implied by the describe() part, I'd just state: 

"should return a successful response with validated email"

the same goes for the other tests descriptions

Reply:
fix into 648f868